### PR TITLE
feat!: release version 2

### PR DESCRIPTION
**Pull request by jonaslagoni**
## [2.0.0](https://github.com/asyncapi/modelina/compare/v1.9.0...v2.0.0) (2023-10-16)

> v2 generates more accurate types such as unions, and nullable and optional properties

You can read the migration guide here: https://www.modelina.org/docs/version-1-to-2

### Modelina Breaking Changes
These are the potential breaking changes for Modelina itself, that might break the generation code:
* #1505

### Models Breaking Changes
These are the potential breaking changes for the generated models if you upgrade:
* #1498
* #1136
* #1140
* #1141
* #1149
* #1169
* #1243
* #1271
* #1277
* #1281
* #1290
* #1296
* #1439
* #1485

### New Features
* #1469

### Bug Fixes
* #1509
* #1502
* #1283
* #1429
* #1472
* #1486

### Performance Improvements
* #1501 

### Other Changes
* #1269
* #1282

### CODE REVIEW

**Comment by jonaslagoni** [comment]
@Ferror @leigh-johnson @Samridhi-98 @magicmatatjahu @kennethaasan pinging you as after https://github.com/asyncapi/modelina/pull/1112 I plan to release v2, so this is a heads-up in case you have anything you want added beforehand 🙂

Right after the release I am gonna re-create the `next` branch so we can start breaking some more code if necessary 😆

**Review by kennethaasan — DISMISSED** [review]

File: .github/workflows/blackbox-testing.yml [hunk 1]
```diff
@@ -13,11 +13,16 @@ jobs:
         os: [ubuntu-latest]
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
```
Comment (Ferror) [inline .github/workflows/blackbox-testing.yml:16]: FYI 4.1 version already avaialble ;p

**Review by Ferror — APPROVED** [review]

File: .github/workflows/blackbox-testing.yml (hunk 1 above)
Comment (jonaslagoni) [reply .github/workflows/blackbox-testing.yml:16]: It's something we can upgrade later 😄

**Comment by jonaslagoni** [comment]
I am gonna merge this tomorrow 🙂 

I am gonna update next with the latest master changes right before I release it (no need to re-approve) 🙂

**Pull request merged**